Dockerfile

```diff
@@ -4,18 +4,28 @@ FROM node:9.5.0
 RUN mkdir -p /diffy/backend
 RUN mkdir -p /diffy/frontend
 
-# Frontend
+# Frontend dependencies
 COPY ./frontend/package.json /diffy/frontend/
 WORKDIR /diffy/frontend
 RUN npm install
+
 # Angular stuff (cli and dev)
 RUN npm install -g @angular/cli
 
-# Backend
+# Backend dependencies
 COPY ./backend/package.json /diffy/backend/
 WORKDIR /diffy/backend
 RUN npm install
 
+COPY ./backend /diffy/backend/
```

Comment:
This makes sense. I had it like this for a while and then for development convenience I changed it to its current state. I'm struggling a bit with best practices between docker for development and docker for production (if that even makes sense). If you have some advice I'll gladly take it.

Reply:
Yes, I totally understand your pain. I still struggle with it :D .

Like I've mentioned in the comment above, you can create `override` files, that override small pieces of the base docker-compose.yml. I guess there could be the base one, a dev override and a prod override?

Reply:
I've made a small override to return the docker-compose to it's original state:

https://github.com/Qu4tro/diffy/pull/1

I can merge it to this PR, if you want it.

Runs with 
`docker-compose -f docker-compose.yml -f docker-compose.dev.yml [up|down|ps|...]`

Reply:
Yes, I like this idea. Can you do the same for Dockerfile? As in, keep the current one as Dockerfile.dev and have the ideal prod one as Dockerfile.prod? It achieves the same goals and makes it explicit. Could you add them to these PR and I will merge it.

Reply:
Yep, no problem.

The docker dev files now represent the docker setup before this PR.

Is this what had in mind?